wp-content/civi-extensions/goonjcustom/CRM/Goonjcustom/CivirulesAction/CreateEventForContact.php

```diff
@@ -15,7 +15,50 @@ public function processAction(CRM_Civirules_TriggerData_TriggerData $triggerData
 		$endDate = $originalData['custom_74'] ?? null;
 		$contactId = $originalData['contact_id'] ?? null;
 		$activityId = $originalData['activity_id'] ?? null;
-	
+		$location = $originalData['custom_69'] ?? null;
+		$district = $originalData['custom_72'] ?? null;
+		$state = $originalData['custom_71'] ?? null;
+		$country = $originalData['custom_70'] ?? null;
+
+		// Save an address for the contact
+		try {
+			$addressResult = \Civi\Api4\Address::create(FALSE)
+				->addValue('contact_id', $contactId)
+				->addValue('street_address', $location)
+				->addValue('city', $district)
+				->addValue('state_province_id', $state)
+				->addValue('country_id', $country)
+				->setFixAddress(FALSE)
+				->execute();
+
+			$addressData = $addressResult->first();
+			$addressId = $addressData['id'] ?? null;
+
+			if ($addressId === null) {
+				throw new Exception('Address ID not found.');
```

Comment:
in which case these exceptions will be thrown? what will be the impact for the end user?

Reply:
I think we don't need this because it does not impact the end user.